### ISSUE TITLE
config autogeneration not working in Meson or Make

### ISSUE THREAD

**Issue by mmols** [pgbackrest/pgbackrest#2382]
I'm looking at adding support for #2023, but having some issues with auto-generation associated with the required config updates (`parse.auto.c.inc`, `config.auto.h`). Using either meson or Make, I don't see these files being re-generated with changes to `config.yaml`.

The smallest change to reproduce this would be to add `auto` to `repo-azure-key-type` and build from there:

https://github.com/pgbackrest/pgbackrest/compare/main...pgEdge:pgbackrest:azure-managed-identities#diff-0d570e156ecb6478ecc163e6abbe3917d9284de7ebb71011e16b083e0badec0aL1864

The files are not re-generated and you end up with:

```
PGBACKREST_REPO1_AZURE_ACCOUNT=foo PGBACKREST_REPO1_AZURE_KEY=bar ./src/pgbackrest info --repo1-type=azure --repo1-azure-key-type=auto --repo1-azure-container=baz
ERROR: [032]: 'auto' is not allowed for 'repo1-azure-key-type' option
```

The contribution guide says:

```
Options can be added to a command or multiple commands. Options can be configuration file only, command-line only or valid for both. Once an option is successfully added, the `config.auto.h` and `parse.auto.c.inc` files will automatically be generated by the build system.
```

Is there another way to trigger this generation? I tried tweaking the Meson configuration with no success. I'm admittedly not very experienced with C or Meson, so I may be making a misstep elsewhere or it may be an issue with my setup.

This is on rockylinux 9 in Docker - I'll comment with setup details.

**Comment by mmols** [comment]
Sample Dockerfile:

```
FROM rockylinux:9.2

ARG TARGETARCH

RUN dnf install -y epel-release
RUN dnf config-manager --set-enabled crb

RUN dnf install -y --allowerasing \
    wget \
    make \
    gcc \
    openssl-devel \
    libxml2-devel \
    lz4-devel \
    libzstd-devel \
    bzip2-devel \
    libssh2-devel \
    postgresql-server \
    glibc-all-langpacks \
    libpq-devel \
    libyaml-devel \
    meson

    
RUN mkdir -p /build && wget -q -O - \
    https://github.com/pgedge/pgbackrest/archive/azure-managed-identities.tar.gz | \
    tar zx -C /build

RUN meson setup /build/pgbackrest-azure-managed-identities /build/pgbackrest-azure-managed-identities/build

WORKDIR /build/pgbackrest-azure-managed-identities/build
```

To recreate:
```
docker exec -it <built-sha>
ninja
PGBACKREST_REPO1_AZURE_ACCOUNT=foo PGBACKREST_REPO1_AZURE_KEY=bar ./src/pgbackrest info --repo1-type=azure --repo1-azure-key-type=auto --repo1-azure-container=baz
```

**Comment by dwsteele** [comment]
> The files are not re-generated and you end up with

Yeah, the contributing guide was off here. Code generation for config happens when test.pl is run, not at compile time. I have fixed that here: b9a9ef2d5.

In your case you'll want to run the Azure tests anyway so:
```
pgbackrest/test/test.pl --module=storage --test=azure
```
Will perform code generation and run the test.
